### PR TITLE
Categorize Event Reference events

### LINKED ISSUE #455: Add categories for event reference (docs)
### Summary

Put events in sub-categories in the event-reference in `api.html`

### What is the feature request for?

The documentation

### The Problem

It's kind of hard not knowing which events fall in which categories, I feel it would be easier if they were in sub-categories.

### The Ideal Solution

Make subcategories for each type of events like `Roles`, `Messages`, etc

Like

```rst
Event Reference
---------------------------

Message Event Reference
~~~~~~~~~~~~~~~

.. autofunction:: on_message
.. autofunction:: on_message_delete
.. autofunction:: on_message_edit
```

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by pandaninjas**
## Summary

This pull requests resolves #455. This classifies events in the documentation. Let me know if you want anything changed

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by plun1331** [comment]
The diff I'm looking at displays the entire API reference as modified, and as such I can't really see what changes were made.

**Comment by pandaninjas** [comment]
> The diff I'm looking at displays the entire API reference as modified, and as such I can't really see what changes were made.
I'm going to see what I can do about that.

**Comment by pandaninjas** [comment]
Fixed in latest commit

**Comment by baronkobama** [comment]
I feel at the very least these categories should be alphabetically sorted as these sections feel kinda messy. Atop of that, I'm not sure if having an "other" section is a great idea and it'd probably be better to just be more specific in each section's title.

**Comment by pandaninjas** [comment]
> I feel at the very least these categories should be alphabetically sorted as these sections feel kinda messy. Atop of that, I'm not sure if having an "other" section is a great idea and it'd probably be better to just be more specific in each section's title.

For the other section, do you want me to create a section for each event in other?

File: docs/api.rst [hunk 1]
```diff
@@ -401,6 +404,8 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param shard_id: The shard ID that has disconnected.
     :type shard_id: :class:`int`
 
+Bot State Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~
 .. function:: on_ready()
```
Comment (plun1331) [inline docs/api.rst:409]: This can be categorized as 'connection'

File: docs/api.rst [hunk 2]
```diff
@@ -678,6 +693,8 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param payload: The raw event payload data.
     :type payload: :class:`RawMessageUpdateEvent`
 
+Reaction Event Reference
```
Comment (plun1331) [inline docs/api.rst:696]: This can be categorized under 'mesaage'

File: docs/api.rst [hunk 3]
```diff
@@ -568,6 +579,8 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param message: The current message.
     :type message: :class:`Message`
 
+Message Deletion Event Reference
```
Comment (plun1331) [inline docs/api.rst:582]: These subcategories are redundant and just add clutter

File: docs/api.rst [hunk 4]
```diff
@@ -1457,6 +1429,81 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param payload: The event's data.
     :type payload: :class:`AutoModActionExecutionEvent`
 
+Other Events Reference
+~~~~~~~~~~~~~~~~~~~~~~
+.. function:: on_interaction(interaction)
```
Comment (plun1331) [inline docs/api.rst:1434]: This and on_application_command (as well as other things like autocomplete that aren't documented) can be put in an 'interactions' category

File: docs/api.rst [hunk 5]
```diff
@@ -1457,6 +1429,81 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+              on_group_remove(channel, user)
+
+    Called when someone joins or leaves a :class:`GroupChannel`.
+
+    :param channel: The group that the user joined or left.
+    :type channel: :class:`GroupChannel`
+    :param user: The user that joined or left.
+    :type user: :class:`User`
+
+.. function:: on_voice_state_update(member, before, after)
```
Comment (plun1331) [inline docs/api.rst:1487]: Could possibly be placed under members

File: docs/api.rst [hunk 6]
```diff
@@ -366,75 +366,173 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    .. versionadded:: 2.0
 
-    Called when the client is done preparing the data received from Discord. Usually after login is successful
-    and the :attr:`Client.guilds` and co. are filled up.
+    :param interaction: The interaction associated to the unknown command.
+    :type interaction: :class:`Interaction`
 
-    .. warning::
+AutoMod Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:411]:
```suggestion
AutoMod
~~~~~~~
```

File: docs/api.rst [hunk 7]
```diff
@@ -366,75 +366,173 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    Called when an auto moderation action is executed.
+
+    The bot must have :attr:`~Permissions.manage_guild` to receive this, and 
+    :attr:`Intents.auto_moderation_execution` must be enabled.
+
+    :param payload: The event's data.
+    :type payload: :class:`AutoModActionExecutionEvent`
+
+Bans Event Reference
+~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:453]:
```suggestion
Bans
~~~~
```

File: docs/api.rst [hunk 8]
```diff
@@ -366,75 +366,173 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+    This requires :attr:`Intents.bans` to be enabled.
+
+    :param guild: The guild the user got unbanned from.
+    :type guild: :class:`Guild`
+    :param user: The user that got unbanned.
+    :type user: :class:`User`
+
+Channel Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:479]:
```suggestion
Channels
~~~~~~~~
```

File: docs/api.rst [hunk 9]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    .. versionadded:: 1.4
 
-    If the typing event is occurring in a guild,
-    the member that started typing can be accessed via :attr:`RawTypingEvent.member`
+    :param shard_id: The shard ID that has resumed.
+    :type shard_id: :class:`int`
 
-    This requires :attr:`Intents.typing` to be enabled.
+Guild Event Reference
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:638]:
```suggestion
Guilds
~~~~~~
```

File: docs/api.rst [hunk 10]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
-    Since the data payload can be partial, care must be taken when accessing stuff in the dictionary.
-    One example of a common case of partial data is when the ``'content'`` key is inaccessible. This
-    denotes an "embed" only edit, which is an edit in which only the embeds are updated by the Discord
-    embed server.
+    :param channel: The channel that had its webhooks updated.
+    :type channel: :class:`abc.GuildChannel`
 
-    This requires :attr:`Intents.messages` to be enabled.
+Integration Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:756]:
```suggestion
Integrations
~~~~~~~~~~~~
```

File: docs/api.rst [hunk 11]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    .. versionadded:: 2.0
 
-    .. note::
+    :param payload: The raw event payload data.
+    :type payload: :class:`RawIntegrationDeleteEvent`
 
-        Consider using :func:`on_raw_reaction_remove` if you need this and do not want
-        to enable the members intent.
+Interactions Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:802]:
```suggestion
Interactions
~~~~~~~~~~~~
```

File: docs/api.rst [hunk 12]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    :param interaction: The interaction data.
+    :type interaction: :class:`Interaction`
 
-    Called when a message has all its reactions removed from it. Similar to :func:`on_message_edit`,
-    if the message is not found in the internal message cache, then this event
-    will not be called. Consider using :func:`on_raw_reaction_clear` instead.
-
-    This requires :attr:`Intents.reactions` to be enabled.
+Invites Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:821]:
```suggestion
Invites
~~~~~~~
```

File: docs/api.rst [hunk 13]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    This requires :attr:`Intents.invites` to be enabled.
 
-    .. versionadded:: 1.3
+    :param invite: The invite that was deleted.
+    :type invite: :class:`Invite`
 
-    :param payload: The raw event payload data.
-    :type payload: :class:`RawReactionClearEmojiEvent`
+Member/User Event Reference
+~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:860]:
```suggestion
Members/Users
~~~~~~~~~~~~~
```

File: docs/api.rst [hunk 14]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
-    :param after: The updated group channel's new info.
-    :type after: :class:`GroupChannel`
+    :param before: The updated user's old info.
+    :type before: :class:`User`
+    :param after: The updated user's updated info.
+    :type after: :class:`User`
 
-.. function:: on_private_channel_pins_update(channel, last_pin)
+Message Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:946]:
```suggestion
Messages
~~~~~~~~
```

File: docs/api.rst [hunk 15]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
-.. function:: on_integration_create(integration)
-
-    Called when an integration is created.
-
-    This requires :attr:`Intents.integrations` to be enabled.
+    :type payload: :class:`RawMessageUpdateEvent`
 
-    .. versionadded:: 2.0
+Reaction Event Reference
+````````````````````````
```
Comment (Middledot) [inline docs/api.rst:1075]:
```suggestion
Reactions
`````````\
```
ignore the last backslash github breaks without it

File: docs/api.rst [hunk 16]
```diff
@@ -467,879 +565,631 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
 
-    :param channel: The group that the user joined or left.
-    :type channel: :class:`GroupChannel`
-    :param user: The user that joined or left.
-    :type user: :class:`User`
+    :param payload: The raw event payload data.
+    :type payload: :class:`RawReactionClearEmojiEvent`
 
+Scheduled Events Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:1192]:
```suggestion
Scheduled Events
~~~~~~~~~~~~~~~~
```

File: docs/api.rst [hunk 17]
```diff
@@ -1417,45 +1267,231 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+        This is only for the messages sent from the client
+        WebSocket. The voice WebSocket will not trigger this event.
+
+    :param payload: The message that is about to be passed on to the
+                    WebSocket library. It can be :class:`bytes` to denote a binary
+                    message or :class:`str` to denote a regular text message.
+
+Stage Channel Event Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:1324]:
```suggestion
Stage Instances
~~~~~~~~~~~~~~~~
```

File: docs/api.rst [hunk 18]
```diff
@@ -1417,45 +1267,231 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+    .. versionadded:: 2.0
+
+    :param before: The stage instance before the update.
+    :type before: :class:`StageInstance`
+    :param after: The stage instance after the update.
+    :type after: :class:`StageInstance`
+
+Thread Event Reference
+~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:1352]:
```suggestion
Threads
~~~~~~~
```

File: docs/api.rst [hunk 19]
```diff
@@ -1417,45 +1267,231 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+    .. versionadded:: 2.0
+
+    :param before: The updated thread's old info.
+    :type before: :class:`Thread`
+    :param after: The updated thread's new info.
+    :type after: :class:`Thread`
+
+Typing Event Reference
+~~~~~~~~~~~~~~~~~~~~~~
```
Comment (Middledot) [inline docs/api.rst:1450]:
```suggestion
Typing
~~~~~~
```

File: docs/api.rst [hunk 20]
```diff
@@ -1417,45 +1267,231 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+.. function:: on_group_join(channel, user)
+              on_group_remove(channel, user)
+
+    Called when someone joins or leaves a :class:`GroupChannel`.
+
+    :param channel: The group that the user joined or left.
+    :type channel: :class:`GroupChannel`
+    :param user: The user that joined or left.
+    :type user: :class:`User`
```
Comment (Middledot) [inline docs/api.rst:1494]: This can be removed. The event was removed but the documentation stayed

**Review by Middledot — COMMENTED** [review]
I think adding `event reference` after all the subheaders is a bit too cluttery. Also connection and socket events could fit into one category

File: docs/api.rst (hunk 20 above)
Comment (pandaninjas) [reply docs/api.rst:1494]: Good to know

File: docs/api.rst [hunk 21]
```diff
@@ -1417,45 +1318,178 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
```
Comment (Lulalaby) [inline docs/api.rst:1492]:
```suggestion
```

**Review by Lulalaby — DISMISSED** [review]

File: docs/api.rst [hunk 22]
```diff
@@ -366,6 +366,205 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    This requires :attr:`Intents.bans` to be enabled.
+
+    :param guild: The guild the user got unbanned from.
+    :type guild: :class:`Guild`
+    :param user: The user that got unbanned.
+    :type user: :class:`User`
+
+Channels
+~~~~~~~~
+.. function:: on_private_channel_update(before, after)
```
Comment (EmreTech) [inline docs/api.rst:480]: g is before p in the alphabet

File: docs/api.rst [hunk 23]
```diff
@@ -400,7 +600,6 @@ to handle it, which defaults to print a traceback and ignoring the exception.
 
     :param shard_id: The shard ID that has disconnected.
     :type shard_id: :class:`int`
-
```
Comment (EmreTech) [inline docs/api.rst:403]: Why was this line removed?

**Review by EmreTech — COMMENTED** [review]
Few things I spotted, honestly this PR would be a pain to review because we need to make sure everything is sorted correctly according to the alphabet

**Review by EmreTech — DISMISSED** [review]
Sorry, I misunderstood the purpose of this PR

File: docs/api.rst [hunk 24]
```diff
@@ -366,6 +366,205 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+    Called when an application command has an error.
+
+    .. versionadded:: 2.0
+
+    :param context: The ApplicationContext associated to the command that has an error.
+    :type context: :class:`ApplicationContext`
+
+    :param exception: The DiscordException associated to the error.
+    :type exception: :class:`DiscordException`
```
Comment (plun1331) [inline docs/api.rst:399]: Shouldn't this be an ApplicationCommandInvokeError?

File: docs/api.rst [hunk 25]
```diff
@@ -518,281 +685,172 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
 
-    .. note::
+.. function:: on_guild_emojis_update(guild, before, after)
 
-        This doesn't require :attr:`Intents.members` within a guild context,
-        but due to Discord not providing updated user information in a direct message
-        it's required for direct messages to receive this event.
-        Consider using :func:`on_raw_reaction_add` if you need this and do not otherwise want
-        to enable the members intent.
+    Called when a :class:`Guild` adds or removes :class:`Emoji`.
```
Comment (plun1331) [inline docs/api.rst:760]:
```suggestion
    Called when a :class:`Guild` adds or removes an :class:`Emoji`.
```

File: docs/api.rst [hunk 26]
```diff
@@ -518,281 +685,172 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
-    Due to the inherently raw nature of this event, the data parameter coincides with
-    the raw data given by the `gateway <https://discord.com/developers/docs/topics/gateway#message-update>`_.
+.. function:: on_guild_role_create(role)
+              on_guild_role_delete(role)
 
-    Since the data payload can be partial, care must be taken when accessing stuff in the dictionary.
-    One example of a common case of partial data is when the ``'content'`` key is inaccessible. This
-    denotes an "embed" only edit, which is an edit in which only the embeds are updated by the Discord
-    embed server.
+    Called when a :class:`Guild` creates or deletes a new :class:`Role`.
```
Comment (plun1331) [inline docs/api.rst:738]:
```suggestion
    Called when a :class:`Guild` creates or deletes a :class:`Role`.
```

File: docs/api.rst [hunk 27]
```diff
@@ -518,281 +685,172 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    :type guild: :class:`Guild`
 
-    This requires :attr:`Intents.messages` to be enabled.
+.. function:: on_guild_update(before, after)
 
-    :param before: The previous version of the message.
-    :type before: :class:`Message`
-    :param after: The current version of the message.
-    :type after: :class:`Message`
+    Called when a :class:`Guild` updates, for example:
```
Comment (plun1331) [inline docs/api.rst:721]:
```suggestion
    Called when a :class:`Guild` is updated, for example:
```

File: docs/api.rst [hunk 28]
```diff
@@ -518,281 +685,172 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+    :type before: Sequence[:class:`Emoji`]
+    :param after: A list of emojis after the update.
+    :type after: Sequence[:class:`Emoji`]
 
-    Called when a message has a reaction added. Unlike :func:`on_reaction_add`, this is
-    called regardless of the state of the internal message cache.
+.. function:: on_guild_stickers_update(guild, before, after)
 
-    This requires :attr:`Intents.reactions` to be enabled.
+    Called when a :class:`Guild` updates its stickers.
```
Comment (plun1331) [inline docs/api.rst:773]: Consistency

```suggestion
    Called when a :class:`Guild` adds or removes a sticker.
```

File: docs/api.rst [hunk 29]
```diff
@@ -810,248 +868,47 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
-
-    :param integration: The integration that was created.
-    :type integration: :class:`Integration`
+Invites
+~~~~~~~
+.. function:: on_invite_create(invite)
 
-.. function:: on_integration_update(integration)
+    Called when an :class:`Invite` is created.
+    You must have the :attr:`~Permissions.manage_channels` permission to receive this.
```
Comment (plun1331) [inline docs/api.rst:876]: I don't actually know what permission is required for this but `manage_channels` doesn't make a whole lot of sense to me

File: docs/api.rst (hunk 24 above)
Comment (pandaninjas) [reply docs/api.rst:399]: I don't know, I copied and pasted it from the previous, and in current pycord docs, it is a `DiscordException`
https://docs.pycord.dev/en/master/api.html#discord.on_application_command_error

File: docs/api.rst (hunk 23 above)
Comment (pandaninjas) [reply docs/api.rst:403]: Typo, probably

**Comment by Lulalaby** [comment]
@pandaninjas please resolve conflicts
@Pycord-Development/maintainers please review

**Comment by pandaninjas** [comment]
Oops, I accidentally discarded all my commits

**Comment by pandaninjas** [comment]
I can't seem to make GitHub recognize the new commits on this branch, I've opened a new PR #1644
